src/components/map/Map.js

```diff
@@ -6,68 +6,72 @@ import { leafletSearch } from 'leaflet-search'; // eslint-disable-line no-unused
 import 'leaflet/dist/leaflet.css';
 import 'leaflet-draw/dist/leaflet.draw.css';
 import 'leaflet-search/dist/leaflet-search.min.css';
-import { MAP_MIN_ZOOM, MAP_INITIAL_ZOOM, MAP_CENTER, BASEMAP_ATTRIBUTION, BASEMAP_TILE, DRAW_CONTROL } from '../../constants/map';
+import { MAP_CONFIG, DRAW_CONTROL } from '../../constants/map';
 
 class Map extends React.Component {
 
   componentDidMount() {
     this.initMap();
-    this.initLayers();
-    if (this.props.editable) {
-      this.initDrawing();
-    }
   }
 
   componentWillUnmount() {
     this.remove();
   }
 
-  onDrawEventComplete(e) {
-    const layer = e.layer;
-    this.featureGroup.addLayer(layer);
-    this.props.onDrawComplete && this.props.onDrawComplete(this.featureGroup.getLayers()[0].toGeoJSON());
-  }
-
-  onDrawEventDelete(e) {
-    const layer = e.layer;
-    this.featureGroup.removeLayer(layer);
-    this.props.onDrawComplete && this.props.onDrawComplete();
-    if (this.featureGroup.getLayers().length === 0) {
-      this.enablePolygonDraw();
-    }
+  componentWillReceiveProps(nextProps) {
+    // Zoom
+    this.map.setZoom(nextProps.mapConfig.zoom);
   }
 
   initMap() {
     this.map = L.map('map', {
-      minZoom: MAP_MIN_ZOOM,
-      zoom: MAP_INITIAL_ZOOM,
-      center: MAP_CENTER,
-      detectRetina: true
+      minZoom: MAP_CONFIG.minZoom,
+      zoom: this.props.mapConfig.zoom,
+      center: MAP_CONFIG.center,
+      detectRetina: true,
+      zoomControl: isNaN(this.props.mapConfig.zoomControl) ? MAP_CONFIG.zoomControl : this.props.mapConfig.zoomControl,
```

Comment:
This was an abstraction to make the component more reusable later on. I prefer this incase someone puts in a faulty value for the setting. Yes, we can make it simpler with value || false.